craft_application/commands/lifecycle.py

```diff
@@ -422,6 +424,29 @@ def _run_real(
         if shell_after:
             _launch_shell()
 
+    @staticmethod
+    def _normalize_paths(
```

Comment:
I'm not sure I love this name, but I can't think of anything better. The closest I can think of is "relativize" which simply moves the problem from "what's normal?" to "relative to what?"

If anyone else comes up with an agreeable name I'd appreciate that, but otherwise I'm happy to leave it as-is.

Reply:
Also this function is silently filtering paths, which isn't what I would expect it to do based on the name.

Reply:
We should raise an error instead of silently ignoring files. I left that to #770, but we can do that now. Also `relativize_paths` sounds reasonable since we're passing the root path.